src/expr/src/relation/mod.rs

```diff
@@ -1962,6 +1964,53 @@ pub struct RowSetFinishing {
     pub project: Vec<usize>,
 }
 
+impl Explain<'_> for RowSetFinishing {
+    type Context = ();
+
+    type Text = Self;
+
+    type Json = Self;
+
+    type Dot = UnsupportedFormat;
+
+    fn explain_text(
+        &mut self,
+        _config: &mz_repr::explain_new::ExplainConfig,
+        _context: &Self::Context,
+    ) -> Result<Self::Text, mz_repr::explain_new::ExplainError> {
+        Ok(self.clone())
+    }
+
+    fn explain_json(
+        &mut self,
+        _config: &mz_repr::explain_new::ExplainConfig,
+        _context: &Self::Context,
+    ) -> Result<Self::Json, mz_repr::explain_new::ExplainError> {
+        Ok(self.clone())
+    }
+}
+
+impl DisplayText for RowSetFinishing {
+    fn fmt_text(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(f, "row_set_finishing")?;
+        writeln!(f, "\torder_by: [{}]", bracketed("[", "]", separated(",", &self.order_by)))?;
+        if let Some(limit) = self.limit {
+            writeln!(f, "\tlimit: {}", limit)?;
+        }
+        if self.offset > 0 {
+            writeln!(f, "\toffset: {}", self.offset)?;
+        }
+        // TODO: add an option for column ranges?
+        writeln!(f, "\toutput: {}", bracketed("[", "]", separated(",", &self.project)))
```

Comment:
This was motivated by the fact that postgresql calls it "output" columns.
```
explain verbose select * from t1 order by f1 limit 5;
                               QUERY PLAN                                
-------------------------------------------------------------------------
 Limit  (cost=70.14..70.15 rows=5 width=8)
   Output: f1, f2
   ->  Sort  (cost=70.14..75.79 rows=2260 width=8)
         Output: f1, f2
         Sort Key: t1.f1
         ->  Seq Scan on public.t1  (cost=0.00..32.60 rows=2260 width=8)
               Output: f1, f2
```

That said, output in postgresql is a combination of map and project à la QGM, whereas here it is only a column projection.

Reply:
Thought about it, and the reason it's only a column projection here is that we need potentially project away the order by columns. We don't want to add map functionality here because `RowSetFinishing` happens only on a single thread.

Reply:
> This was motivated by the fact that postgresql calls it "output" columns.

OK, maybe it is a good idea to try to use the `explain verbose` output as a tie-braker if unsure. I'm happy to stick with that and will rework `HIR` and `MIR` accordingly when I come to them.